### PR TITLE
allow clicking the back button from a PageHeader driver

### PR DESCRIPTION
**Pull request by dor-itzhaki**
### What changed
- PageHeader driver exposes "clickBackButton" function
- PageHeader puts a dataHook on the Button component instead of its wrapper
- PageHeader "onBackClick" prop is now tested

### Why it changed
- The "onBackClick" prop was not tested
- Driver was this lacking functionality.

---

- [ ] Change is tested
- [ ] Change is documented
- [ ] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

File: src/PageHeader/PageHeader.js [hunk 1]
```diff
@@ -147,9 +147,12 @@ export default class PageHeader extends WixComponent {
                     [s.minimized]: minimized,
                     [s.darkTheme]: isDarkTheme(hasBackgroundImage, minimized),
                   })}
-                  data-hook="page-header-backbutton"
                 >
-                  <Button onClick={onBackClicked} theme="icon-white">
+                  <Button
+                    dataHook="page-header-backbutton"
```
Comment (dor-itzhaki) [inline src/PageHeader/PageHeader.js]: i see that the norm is to set the data-hook on the wrapper component, but couldn't really understand why. in my opinion, putting the data-hook on the wrapper only requires the driver later on to be familiar with the internals of the used component. BUT, if that's an issue, i could move the data-hook back and just look for an inner "button" in the driver.

File: src/PageHeader/PageHeader.spec.js [hunk 2]
```diff
@@ -169,6 +170,21 @@ describe('PageHeader', () => {
     expect(driver.isBackButtonExists()).toBeFalsy();
   });
 
+  it('should execute the given back button callback once the back button is clicked', () => {
+    const backButtonCallback = sinon.spy();
```
Comment (hepiyellow) [inline src/PageHeader/PageHeader.spec.js]: Please use `jest.fn()`

Comment (hepiyellow) [inline src/PageHeader/PageHeader.spec.js]:
```suggestion
    const backButtonCallback = jest.fn();
```

File: src/PageHeader/PageHeader.spec.js [hunk 3]
```diff
@@ -169,6 +170,21 @@ describe('PageHeader', () => {
...
+      <PageHeader
+        title={title}
+        showBackButton
+        onBackClicked={backButtonCallback}
+      />
+    );
+    const driver = createDriver(pageHeader);
+    driver.clickBackButton();
+
+    expect(backButtonCallback.calledOnce).toBeTruthy();
```
Comment (hepiyellow) [inline src/PageHeader/PageHeader.spec.js]:
```suggestion
    expect(backButtonCallback).toBeCalledTimes(1);
```

File: src/PageHeader/PageHeader.spec.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ import { createDriverFactory } from 'wix-ui-test-utils/driver-factory';
 import Button from '../Button';
 import Breadcrumbs from '../Breadcrumbs/Breadcrumbs';
 import PropTypes from 'prop-types';
+import sinon from 'sinon';
```
Comment (hepiyellow) [inline src/PageHeader/PageHeader.spec.js]:
```suggestion
```

**Pull request merged**